my_methods.rb

```diff
@@ -0,0 +1,182 @@
+# rubocop: disable Metrics/ModuleLength
+# rubocop: disable Metrics/PerceivedComplexity, Metrics/CyclomaticComplexity
+
+module Enumerable
+  def my_each
+    return to_enum(:my_each) unless block_given?
+
+    each do |num|
+      yield(num)
+    end
+  end
+
+  def my_each_with_index
+    return to_enum(:my_each_with_index) unless block_given?
+
+    i = 0
+    my_each do |num|
+      yield(num, i)
+      i += 1
+    end
+  end
+
+  def my_select
+    return to_enum(:my_select) unless block_given?
+
+    if is_a?(Array)
+      arr = []
+      my_each do |num|
+        arr << num if yield(num)
+      end
+      arr
+    elsif is_a?(Hash)
+      hash = {}
+      my_each do |key, val|
+        hash[key] = val if yield(key, val)
+      end
+      hash
+    end
+  end
+
+  def my_all?(var = nil)
+    unless var.nil?
+      if var.is_a?(Regexp)
+        my_each { |val| return false unless val.match(var) }
+      elsif var.is_a?(Module)
+        my_each { |val| return false unless val.is_a?(var) }
+      else
+        my_each { |val| return false if val != var }
+      end
+      return true
+    end
+    unless block_given?
+      my_each { |val| return false if val == false || val.nil? }
+      return true
+    end
+    my_each do |num|
+      return false unless yield(num)
+    end
+    true
+  end
+
+  def my_any?(var = nil)
+    unless var.nil?
+      if var.is_a?(Regexp)
+        my_each { |val| return true if val.match(var) }
+      elsif var.is_a?(Module)
+        my_each { |val| return true if val.is_a?(var) }
+      else
+        my_each { |val| return true if val == var }
+      end
+      return false
+    end
+    unless block_given?
+      my_each { |val| return true if val != false && !val.nil? }
+      return false
+    end
+
+    my_each do |num|
+      return true if yield(num)
+    end
+    false
+  end
+
+  def my_none?(var = nil)
+    unless var.nil?
+      if var.is_a?(Regexp)
+        my_each { |val| return false if val.match(var) }
+      elsif var.is_a?(Module)
+        my_each { |val| return false if val.is_a?(var) }
+      else
+        my_each { |val| return false if val == var }
+      end
+      return true
+    end
+    unless block_given?
+      my_each { |val| return false if val != false && !val.nil? }
+      return true
+    end
+
+    my_each do |num|
+      return false if yield(num)
+    end
+    true
+  end
```

Comment:
[OPTIONAL]

I think we could get the block of my_none? from my_any? we should only do something like !my_any? for my_none? to work. For example:

```ruby
def my_none?(arg = nil, &block)
!my_any?(arg, &block)
end

```
To make this work we have to use block.call(element) instead of yield(element) in my_any?